compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1564,15 +1764,16 @@ public void visit(BLangLiteral astLiteralExpr) {
 
     @Override
     public void visit(BLangMapLiteral astMapLiteralExpr) {
-        visitTypedesc(astMapLiteralExpr.pos, astMapLiteralExpr.getBType(), Collections.emptyList());
+        BType type = astMapLiteralExpr.getBType();
+        createNewTypedescInst(type, astMapLiteralExpr.pos);
```

Comment:
Why is this here? Wasn't the whole point moving typedesc construction to when resolving the type(desc)?

Reply:
If this is for the readonly field scenario, do a specific check here, rather than within `createNewTypedescInst`.

Reply:
The reason for creating a 'typedesc' instruction is to differentiate a map from a record. We need that information when generating typedesc instructions. (For maps we need to create `TypedescValueImpl`).

Reply:
My question is about why we are doing that when visiting the mapping constructor. Wasn't the conclusion that the typedesc has to be created when visiting the typedesc (except with readonly fields)? 

> We need that information when generating typedesc instructions. (For maps we need to create `TypedescValueImpl`).

Why can't we do this when visiting the type/typdesc?

Reply:
```ballerina
function value() {
    map<record{}> k;
}
```

You mean to generate the type descriptor(typedesc) when visiting the type node, right? In the above scenario, we need to create the type descriptor(typedesc) with a  `map<record{}>` type. In the generated constrained type node, the type is a map and constraint is record. Is there a way to obtain that map type(map<record{}>) from the constraint type node?

Reply:
In the generated constrained type node, the type is a map and constraint is record

Isn't the map type here the type you are looking for?

Reply:
```ballerina
function value() {
    map<record{int a = 10;}> k = {};
}
```
No, In this example type of constraint type node is set to builtin ref type(builtin Map type) but the type of the expression is `map<record {| int a; anydata...; |}>`. We need to create typedesc for `map<record {| int a; anydata...; |}>` type.

Reply:
What about `org.wso2.ballerinalang.compiler.tree.BLangNode#determinedType`? Please create an issue for the type being the builtin type.

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/39288

Reply:
I assume we can use `determinedType` for now.